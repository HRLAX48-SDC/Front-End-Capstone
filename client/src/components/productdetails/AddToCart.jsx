import React from 'react';
import Button from 'react-bootstrap/Button';

function AddToCart(props) {
  return (
    <Button variant='dark' type='button' className='btn btn-default btn-sm p_flex_child_cart' onClick={props.handleClick}>
      <b> Add to Cart +</b>
    </Button>
  );
}

export default AddToCart;