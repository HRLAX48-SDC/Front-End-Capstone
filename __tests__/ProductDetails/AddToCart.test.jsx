import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { configure, mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import AddToCart from '../../client/src/components/productdetails/AddToCart.jsx';

describe('Add to cart', () => {
  it('Submit button exists', () => {
    const testAddToCart = renderer.create((
      <AddToCart />
    )).root;

    const submitButtonInstance = testAddToCart.findAllByProps({
      type: 'button',
      variant: 'dark'
    });
    expect(submitButtonInstance.length).toEqual(1);
  })
})

test('renders AddToCart component correctly', () => {
  const addToCart = renderer.create(<AddToCart />).toJSON();
  expect(addToCart).toMatchSnapshot();
});
