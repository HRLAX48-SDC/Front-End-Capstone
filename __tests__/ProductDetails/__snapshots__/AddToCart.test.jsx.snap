// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddToCart component correctly 1`] = `
<button
  className="btn btn-default btn-sm p_flex_child_cart btn btn-dark"
  disabled={false}
  type="button"
>
  <b>
     Add to Cart +
  </b>
</button>
`;
